robotools/evotools/__init__.py

```diff
@@ -181,6 +182,318 @@ def _int_to_tip(tip_int: int):
+    if not len(wells) == len(tips):
+        raise ValueError(f"Invalid wells: wells and tips need to have the same length.")
+    if labware_position is None:
+        raise ValueError("Missing required parameter: position")
+    if not all(isinstance(position, int) for position in labware_position) or any(
+        position < 0 for position in labware_position
+    ):
+        raise ValueError(f"Invalid position: {labware_position}")
+
+    if volume is None:
+        raise ValueError("Missing required parameter: volume")
+    if type(volume) == list:
+        for vol in volume:
+            try:
+                vol = float(vol)
+            except:
+                raise ValueError(f"Invalid volume: {vol}")
+            if vol < 0 or vol > 7158278 or numpy.isnan(vol):
+                raise ValueError(f"Invalid volume: {vol}")
+            if max_volume is not None and vol > max_volume:
+                raise InvalidOperationError(f"Invalid volume: volume of {vol} exceeds max_volume.")
+        if not len(volume) == len(tips) == len(wells):
+            raise Exception(
+                f"Invalid volume: Tips, wells, and volume lists have different lengths ({len(tips)}, {len(wells)} and {len(volume)}, respectively)."
+            )
+    elif type(volume) == float or type(volume) == int:
+        # test volume like in the list section
+        if volume < 0 or volume > 7158278 or numpy.isnan(volume):
+            raise ValueError(f"Invalid volume: {volume}")
+        if max_volume is not None and volume > max_volume:
+            raise InvalidOperationError(f"Invalid volume: volume of {volume} exceeds max_volume.")
+        # convert volume to list and multiply list to reach identical length as wells
+        volume = [float(volume)]
+        volume = volume * len(wells)
+    else:
+        raise ValueError(f"Invalid volume: {volume}")
+
+    # apply rounding and corrections for the right string formatting
+    volume = [numpy.round(vol, decimals=2) for vol in volume]
+
+    if liquid_class is None:
+        raise ValueError(f"Missing required parameter: liquid_class")
+    if not isinstance(liquid_class, str) or ";" in liquid_class:
+        raise ValueError(f"Invalid liquid_class: {liquid_class}")
+
+    if tips is None:
+        raise ValueError(f"Missing required parameter: tips")
+    for tip in tips:
+        if type(tip) is not int and type(tip) is not Tip:
+            raise ValueError(f"Invalid type of tips: {type(tip)}. Has to be int or Tip.")
+    tecan_tips = []
+    for tip in tips:
+        if isinstance(tip, int) and not isinstance(tip, Tip):
+            # User-specified integers from 1-8 need to be converted to Tecan logic
+            tip = _int_to_tip(tip)
+        tecan_tips.append(tip)
+
+    if tecan_tips:
+        return labware, wells, labware_position, volume, liquid_class, tecan_tips
+    else:
+        return labware, wells, labware_position, volume, liquid_class, tips
```

Comment:
`if not tecan_tips` is only the case when `tips` was an empty list, but then `tecan_tips == tips == []` so there's no need to discriminate between one and the other

```suggestion
    return labware, wells, labware_position, volume, liquid_class, tecan_tips
```